docs/lsm/attributes_metadata/attributes_metadata.md

```diff
@@ -0,0 +1,340 @@
+    string interface_name dict interface_name__annotations = {"annotation": "value"}
+end
+```
+
+### Annotations on relational attributes
+
+Annotations can be attached to a relational attribute by replacing the `--` part of the relationship definition with an instance
+of the `lsm::RelationAnnotations` entity. This entity has a dict attribute `annotations` that represents the annotations that
+should be set on the relational attribute. The values of this dictionary must be strings. By convention the name of the
+`lsm::RelationAnnotations` instance should be prefixed and suffixed with two underscores. This improves the readability of the
+relationship definition.
+
+The example below illustrates how the annotation `annotation=value` can be attached to the relational attribute `ports`.
+
+```inmanta
+entity Router extends lsm::ServiceEntity:
+    string name
+end
+
+entity Port extends lsm::EmbeddedEntity:
+    number id
+end
+
+__annotations__ = lsm::RelationAnnotations(
+    annotations={"annotation": "value"}
+) Router.ports [0:] __annotations__ Port._router [1]
+```
+
+### Documentation tabs
+
+Annotations can be used to have the web console render the content of one or more attributes in a tab called *Documentation*.
+This can be used to document and explain details of the service to the user based on information in the orchestration model. The
+attribute value is set during a compile. For example, the following LSM service orchestrates container based labs and provides
+an overview of all the different services and devices in the lab.
+
+![documentation tab](documentation.png)
+
+One or more attributes at the root level can be used and they are all rendered in the documentation tab. In the example the
+documentation is defined as follows:
+
+```inmanta
+entity Service extends lsm::ServiceBase:
+    string documentation = "# Lab documentation"
+    lsm::attribute_modifier documentation__modifier = "r"
+    dict documentation__annotations = {
+        "web_presentation": "documentation",
+        "web_title": "Lab overview",
+        "web_icon": "FaInfo",
+    }
+end
+```
+
+1. An attribute of type string needs to be defined. We recommend to give it a default value so that the tab is not empty when
+   the documentation has not yet been generated by the compiler.
+2. The attribute modifier must be `r` which means that it is not defined by the user and only by the orchestrator.
+3. Annotations are used to control how the attirbute is rendered in the web console:
+   - `web_presentation` is set to `documentation` 
+   - Optionally `web_title` is set to the title that is shown in the tab. This title is only used when multiple attributes are
+     defined as documentation. This title will be used on the card that contains the content of this attribute.
+   - Optionally `web_icon` specifies a font awsome icon name. This icon is also only used when multiple attributes are defined
+     as documentation.
```

Comment:
it might be interesting to link to which font awseome set we support?

Reply:
@LukasStordeur do you have a reference on what icons can be used?